src/api/routes/offer.js

```diff
@@ -158,6 +164,11 @@ module.exports = (app) => {
         validators.disable,
         validators.isExistingOffer,
         validators.canDisable,
+        async (req, res, next) => {
+            const offer = await (new OfferService()).getOfferById(req.params.offerId, req.targetOwner, true);
+
+            return companyMiddleware.isNotDisabled(offer.owner)(req, res, next);
+        },
```

Comment:
Extract this to a middleware function (`isOwnerNotDisabled`) and place it in the offer middleware file.

You can then reuse it in the enable route

Reply:
It's a good time to create the offer middleware file (under middleware/, not middleware/validators/)